x-pack/test/apm_api_integration/basic/tests/services/top_services.ts

```diff
@@ -30,46 +37,58 @@ export default function ApiTest({ getService }: FtrProviderContext) {
     });
 
     describe('when data is loaded', () => {
-      before(() => esArchiver.load('8.0.0'));
-      after(() => esArchiver.unload('8.0.0'));
+      before(() => esArchiver.load(archiveName));
+      after(() => esArchiver.unload(archiveName));
 
       it('returns a list of services', async () => {
         const response = await supertest.get(
           `/api/apm/services?start=${start}&end=${end}&uiFilters=${uiFilters}`
         );
-        // sort services to mitigate unstable sort order
-        const services = sortBy(response.body.items, ['serviceName']);
 
-        expect(response.status).to.be(200);
-        expect(services).to.eql([
-          {
-            serviceName: 'client',
-            agentName: 'rum-js',
-            transactionsPerMinute: 2,
-            errorsPerMinute: 2.75,
-            avgResponseTime: 116375,
-            environments: [],
-          },
-          {
-            serviceName: 'opbeans-java',
-            agentName: 'java',
-            transactionsPerMinute: 30.75,
-            errorsPerMinute: 4.5,
-            avgResponseTime: 25636.349593495936,
-            environments: ['production'],
-          },
-          {
-            serviceName: 'opbeans-node',
-            agentName: 'nodejs',
-            transactionsPerMinute: 31,
-            errorsPerMinute: 3.75,
-            avgResponseTime: 38682.52419354839,
-            environments: ['production'],
-          },
-        ]);
-
-        expect(response.body.hasHistoricalData).to.be(true);
-        expect(response.body.hasLegacyData).to.be(false);
+        expect(response.status).to.eql(200, 'Response status should be 200');
+
+        const metricType = t.strict({
+          value: t.number,
+          over_time: t.array(
+            t.type({
+              x: t.number,
+              y: t.union([t.number, t.null]),
+            })
+          ),
+        });
+
+        const serviceType = t.strict({
+          serviceName: t.string,
+          agentName: t.string,
+          transactionsPerMinute: metricType,
+          avgResponseTime: metricType,
+          // the RUM service will not have transaction error data
+          transactionErrorRate: t.union([metricType, t.undefined]),
+          environments: t.array(t.string),
+          // basic license should not have anomaly scores
+          severity: t.undefined,
+        });
+
+        const responseType = t.type({
+          items: t.array(serviceType),
+          hasHistoricalData: t.literal(true),
+          hasLegacyData: t.literal(false),
+        });
+
+        const data = getValueOrThrow(responseType, response.body);
+
+        const rumService = data.items.find((item) => item.agentName === 'rum-js');
+
+        // RUM transactions don't have event.outcome set,
+        // so they should not have an error rate
+        expect(rumService!.transactionErrorRate).to.be(undefined);
```

Comment:
Would these comment still be needed if the test was split into smaller `it()` sections?

Reply:
What is the output if one of the expectations fails, or the type throw? 
Is it easy to see which expectation or type that failed?

Reply:
Generally I find it easier to both read and debug tests if there's only a single assertion per `it` block

Reply:
That's fair, I prefer comments but I can accept that I'm the odd one out.